packages/mini-browser/src/browser/mini-browser-content.ts

```diff
@@ -395,33 +400,95 @@ export class MiniBrowserContent extends BaseWidget {
         pdfContainer.id = `${this.id}-pdf-container`;
         pdfContainer.style.display = 'none';
 
+        contentArea.appendChild(errorBar);
         contentArea.appendChild(transparentOverlay);
         contentArea.appendChild(pdfContainer);
         contentArea.appendChild(loadIndicator);
         contentArea.appendChild(frame);
 
         parent.appendChild(contentArea);
-        return Object.assign(contentArea, { frame, loadIndicator, pdfContainer, transparentOverlay });
+        return Object.assign(contentArea, { frame, loadIndicator, errorBar, pdfContainer, transparentOverlay });
     }
 
     protected createIFrame(): HTMLIFrameElement {
         const frame = document.createElement('iframe');
         const sandbox = (this.props.sandbox || MiniBrowserProps.SandboxOptions.DEFAULT).map(name => MiniBrowserProps.SandboxOptions[name]);
         frame.sandbox.add(...sandbox);
         this.toDispose.push(addEventListener(frame, 'load', this.onFrameLoad.bind(this)));
+        this.toDispose.push(addEventListener(frame, 'error', this.onFrameError.bind(this)));
         return frame;
     }
 
+    protected createErrorBar(): HTMLElement & Readonly<{ message: HTMLElement }> {
+        const errorBar = document.createElement('div');
+        errorBar.classList.add(MiniBrowserContent.Styles.ERROR_BAR);
+        errorBar.style.display = 'none';
+
+        const icon = document.createElement('span');
+        icon.classList.add('fa', 'fa-times-circle');
+        errorBar.appendChild(icon);
+
+        const message = document.createElement('span');
+        errorBar.appendChild(message);
+
+        return Object.assign(errorBar, { message });
+    }
+
     protected onFrameLoad(): void {
+        clearTimeout(this.frameLoadTimeout);
+        this.maybeResetFrameBackground();
         this.hideLoadIndicator();
+        this.hideErrorBar();
+    }
+
+    protected onFrameError(): void {
+        clearTimeout(this.frameLoadTimeout);
+        this.maybeResetFrameBackground();
+        this.hideLoadIndicator();
+        this.showErrorBar('Error: An error occurred while loading this page');
+    }
+
+    protected onFrameTimeout(): void {
+        clearTimeout(this.frameLoadTimeout);
+        this.maybeResetFrameBackground();
+        this.hideLoadIndicator();
+        this.showErrorBar('Error: Loading this page is taking longer than usual');
     }
 
     protected showLoadIndicator(): void {
+        this.loadIndicator.classList.remove('fade-out');
```

Comment:
For the sake of consistency, can you please move this next to the other CSS constants? So if someone reads the code can guess the class is defined in the same CSS file. Thanks!

Reply:
Note, this is inconsistent with the similar `.theia-hidden` class (it's also [not a constant](https://github.com/jankeromnes/theia/blob/2611d94fb04dee4c0f30fca73a26822b59c36ede/packages/core/src/browser/frontend-application.ts#L196)), but I agree to make it a constant in order to make all CSS classes in `mini-browser-content.ts` consistent.